src/scripting/lua_ui.cc

```diff
@@ -326,9 +330,427 @@ int LuaPanel::indicate(lua_State* L) {
 }
 #endif
 
+/* RST
+   .. method:: create_child(table)
+
+      .. versionadded:: 1.2
+
+      Create a UI widget as a child of this panel as specified by the provided table.
+
+      A UI descriptor table contains multiple keys of type :class:`string`. Common properties are:
+
+         * ``"widget"``: **Mandatory**. The type of widget to create. See below for allowed values.
+         * ``"name"``: **Mandatory** for named panels. The internal name of the panel.
+         * ``"x"``: **Optional**. The horizontal offset inside the parent from the left.
+         * ``"y"``: **Optional**. The vertical offset inside the parent from the top.
+         * ``"w"``: **Optional**. The widget's width.
+         * ``"h"``: **Optional**. The widget's height.
+         * ``"tooltip"``: **Optional**. The widget's tooltip.
+         * ``"resizing"``: **Optional**. If the parent component is a Box,
+           the Resizing strategy to use for layouting. Valid values are:
+
+           * ``"align"``: Use the widget's desired size.
+           * ``"fullsize"``: Use the widget's desired depth and the full available breadth.
+           * ``"fillspace"``: Use the widget's desired breadth and the full available depth.
+           * ``"expandboth"``: Use the full available space.
+
+         * ``"align"``: **Optional**. If the parent component is a Box,
+           the Alignment strategy to use for layouting.
+           Valid values are ``"left"``/``"top"``, ``"center"``,
+           and ``"right"``/`"bottom"``.
```

Comment:
I guess `center` is the default? -> Defaults to `center`

If there are other defaults they should be mentioned.